sys/cpp11-compat/include/riot/thread.hpp

```diff
@@ -240,6 +240,10 @@ class thread {
    */
   template <class F, class... Args>
   explicit thread(F&& f, Args&&... args);
+
+  /**
+   * @brief Disallow copy constructor for constant objects.
```

Comment:
This does not only disallow the copy constructor for constant objects. The `const` in the signature prevents mutation of the argument.
> Disallow copy constructor.

This would probably be enough.